docs/core/development/policy/compatibility-policy.rst

```diff
@@ -162,22 +162,29 @@ The most basic thing that can happen between Twisted versions, of course, is tha
 That means that no application may ever rely on, for example, the value of any **func_code** object's **co_code** attribute remaining stable, or the **checksum** of a .py file remaining stable.
 
 **Docstrings** may also change at any time.
-No application code may expect any Twisted class, module, or method's __doc__ attribute to remain the same.
+Applications must not depend on any Twisted class, module, or method's ``__doc__`` attribute to remain the same.
```

Comment:
```suggestion
Applications must not depend on any Twisted class, module, or method's metadata attributes such as ``__doc__``, ``__name__``, ``__module__`` or ``__qualname__`` to remain the same.
```

Reply:
Eh, `__module__`, `__name__` and `__qualname__` should be stable across releases, I think.

I agree with `__annotations__`: the current annotations are rough first estimates and it's likely a significant number of them will be updated as Twisted itself and projects using Twisted increase their static type checking coverage.

Reply:
> Eh, __module__, __name__ and __qualname__ should be stable across releases, I think.

That would disallow moving a function or class to another module, and aliasing it for compatibility purposes

Reply:
Ah, that makes sense. There is a lot of aliasing in Twisted already to export things from private modules, so those metadata fields are probably not very predictable anyway.